### PR TITLE
 yuzu: Improve behavior when clicking on controller box in Controller applet

### PR DESCRIPTION
**Pull request by flodavid**
- Apply changes on Controller configuration of commit 9524d70 to Controller applet
  - Fix regression of this previous commit: Enabling a controller in its tab did not activate previous controllers
  - Refactor to use same names in both classes

### CODE REVIEW

**Review by scottrory1978 — APPROVED** [review]

File: src/yuzu/configuration/configure_input.cpp [hunk 1]
```diff
@@ -175,28 +188,25 @@ void ConfigureInput::Initialize(InputCommon::InputSubsystem* input_subsystem,
     LoadConfiguration();
 }
 
-void ConfigureInput::propagateMouseClickOnPlayers(size_t player_index, bool checked, bool origin) {
-    // Origin has already been toggled
-    if (!origin) {
-        player_connected[player_index]->setChecked(checked);
-    }
+void ConfigureInput::propagatePlayerNumberChanged(size_t player_index, bool checked,
```
Comment (FearlessTobi) [inline src/yuzu/configuration/configure_input.cpp:191]:
```suggestion
void ConfigureInput::PropagatePlayerNumberChanged(size_t player_index, bool checked,
```

File: src/yuzu/applets/qt_controller.cpp [hunk 2]
```diff
@@ -668,6 +675,29 @@ void QtControllerSelectorDialog::UpdateDockedState(bool is_handheld) {
     }
 }
 
+void QtControllerSelectorDialog::propagatePlayerNumberChanged(size_t player_index, bool checked,
```
Comment (FearlessTobi) [inline src/yuzu/applets/qt_controller.cpp:678]:
```suggestion
void QtControllerSelectorDialog::PropagatePlayerNumberChanged(size_t player_index, bool checked,
```

File: src/yuzu/applets/qt_controller.cpp [hunk 3]
```diff
@@ -155,18 +155,25 @@ QtControllerSelectorDialog::QtControllerSelectorDialog(
...
-                for (std::size_t index = 0; index <= i; ++index) {
-                    connected_controller_checkboxes[index]->setChecked(checked);
-                }
-            } else {
-                for (std::size_t index = i; index < NUM_PLAYERS; ++index) {
-                    connected_controller_checkboxes[index]->setChecked(checked);
-                }
+            // Reconnect current controller if it was the last one checked
+            // (player number was reduced by more than one)
+            bool reconnect_first = !checked && i < player_groupboxes.size() - 1 &&
```
Comment (FearlessTobi) [inline src/yuzu/applets/qt_controller.cpp:160]:
```suggestion
            const bool reconnect_first = !checked && i < player_groupboxes.size() - 1 &&
```

File: src/yuzu/applets/qt_controller.cpp [hunk 4]
```diff
@@ -155,18 +155,25 @@ QtControllerSelectorDialog::QtControllerSelectorDialog(
...
+            // Ensures that connecting a controller changes the number of players
+            if (connected_controller_checkboxes[i]->isChecked() != checked) {
+                // Ensures that the players are always connected in sequential order
+                propagatePlayerNumberChanged(i, checked, reconnect_first);
             }
         });
+        connect(connected_controller_checkboxes[i], &QCheckBox::clicked, [this, i](bool checked) {
+            // Reconnect current controller if it was the last one checked
+            // (player number was reduced by more than one)
+            bool reconnect_first = !checked && i < connected_controller_checkboxes.size() - 1 &&
```
Comment (FearlessTobi) [inline src/yuzu/applets/qt_controller.cpp:171]:
```suggestion
            const bool reconnect_first = !checked && i < connected_controller_checkboxes.size() - 1 &&
```

File: src/yuzu/configuration/configure_input.cpp [hunk 5]
```diff
@@ -101,37 +101,50 @@ void ConfigureInput::Initialize(InputCommon::InputSubsystem* input_subsystem,
...
+            // Ensures that connecting a controller changes the number of players
+            if (connected_controller_checkboxes[i]->isChecked() != checked) {
+                // Ensures that the players are always connected in sequential order
+                propagatePlayerNumberChanged(i, checked);
+            }
+        });
+        connect(connected_controller_checkboxes[i], &QCheckBox::clicked, [this, i](bool checked) {
+            // Reconnect current controller if it was the last one checked
+            // (player number was reduced by more than one)
+            bool reconnect_first = !checked && i < connected_controller_checkboxes.size() - 1 &&
```
Comment (FearlessTobi) [inline src/yuzu/configuration/configure_input.cpp:128]:
```suggestion
            const bool reconnect_first = !checked && i < connected_controller_checkboxes.size() - 1 &&
```

**Review by FearlessTobi — COMMENTED** [review]
Some code suggestions.
For the functionality, @german77 please take a look.

**Comment by flodavid** [comment]
Thanks for the suggestions. I applied them to the commit

**Review by german77 — APPROVED** [review]

**Pull request merged**